### PR TITLE
Command Line UI made as per requirements

### PR DESCRIPTION
**Pull request by Metaverse-Nabeel**
### The following changes have been made to this PR:

- Created `app.rb` file that will serve as your console app entry point. 

**It should have methods that do the following:**
- List all books. 
- List all people. 
- Create a person (teacher or student, not a plain Person). 
- Create a book. 
- Create a rental.
- List all rentals for a given person id

- Created a `main.rb` to define the entry point, this will be a method called main that is invoked at the end of your file.

**This method should do the following:** 

- Present the user with a list of options to perform. 
- Lets users choose an option. If needed, ask for parameters for the option. Have a way to quit the app.

### CODE REVIEW

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @Metaverse-Nabeel, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Clean code
✅ Good first attempt

## Required Changes ♻️
_Check the comments under the review._
I couldn't include these comments inline so I had to do it here
[REQUIRED] I couldn't test if your app is working because I couldn't run your code. I tried `ruby main.rb` and `ruby app.rb` but they are not working. Kindly ensure that `ruby main.rb` command works and give a proper description of how to run your app in your README.rb file.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    @books[book_index] if book_index >= 0 && book_index < @books.length
+  end
+
+  def get_person_by_index(person_index)
+    @people[person_index] if person_index >= 0 && person_index < @people.length
+  end
+
+  def create_rental(book, person, date)
+    if book && person
+      rental = Rental.new(date, book, person)
```
Comment (mwanawabangona) [inline app.rb]: - [ ] `required`
- [ ] Good job so far! 👏 
- [ ] However everything is working okay until i try creating rental date i'm getting an error 😞 
- [ ] Please see the screenshot below

image

**Review by mwanawabangona — CHANGES_REQUESTED** [review]
Hi @Metaverse-Nabeel,

Great work on making the changes requested by a previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* ✅  No linter errors :heavy_check_mark:
* ✅  Repo looks professional :briefcase:
* ✅  Good quality commits
* ✅  Good implementing the requested changes



 ## Required Changes ♻️
_Check the comments under the review._

- [ ] I have noticed you have not added a `Gemfile` hence the project giving me an error when trying to install the dependencies using `bundle install` 
- [ ] please add it because its of good ruby practices and its essential to this project
- [ ] see screenshot below

image
 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (Metaverse-Nabeel) [reply app.rb]: Hello @mwanawabangona

I have made the changes and now it is working perfectly fine. Please review it again. I have submitted the request for review.

**Review by AnselemOdims — APPROVED** [review]
### Hi  🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**